lib/base/networking/apis/auth_handler.dart

```diff
@@ -0,0 +1,117 @@
+
+class AuthHandler {
+  static Future<void> saveToken(List<Cookie> cookies) async {
+    for (var cookie in cookies) {
+      if (cookie.name == 'jwt') {
+        SharedPreferences prefs = await SharedPreferences.getInstance();
+        await prefs.setString('jwt', cookie.value);
+
+        // DEBUG: Check cookie
+        String? jwt = prefs.getString('jwt');
+        print('Current jwt: $jwt');
+        return;
+      }
+    }
+
+    // TODO: Handle error when no jwt cookie is found
+    print('No JWT cookie found');
+  }
+
+  static Future<void> basicAuth(
+    String username,
+    String password,
+  ) async {
+    const url = Routes.basicLogin;
+    final cookieJar = CookieJar();
+    final dio = Dio(
+      BaseOptions(
+        followRedirects: false,
+        validateStatus: (status) {
+          return status! < 500;
+        },
+      ),
+    );
+    dio.interceptors.add(CookieManager(cookieJar));
+
+    final formData = FormData.fromMap({
+      'username': username,
+      'password': password,
+    });
+
+    try {
+      await dio.post(
+        url,
+        data: formData,
+      );
+    } catch (e) {
+      // TODO: Handle network errors
+      print('Request error: $e');
+      return;
+    }
+
+    List<Cookie> cookies = await cookieJar.loadForRequest(Uri.parse(url));
+
+    // Save jwt token
+    await saveToken(cookies);
+  }
+
+  static Future<void> ssoAuth(BuildContext context) async {
+    print('ssoAuth started');
+    // Redirect the user to the Shibboleth login page
+    print('Login URL: $Routes.ssoLogin');
+
+    // Open the login page in a web view
+    await Navigator.push(
+      context,
+      MaterialPageRoute(
+        builder: (context) => webview.InAppWebView(
+          initialUrlRequest:
+              webview.URLRequest(url: Uri.parse(Routes.ssoLogin)),
+          onWebViewCreated: (webview.InAppWebViewController controller) {
+            print('Web view created');
+          },
+          onLoadStart: (webview.InAppWebViewController controller, Uri? url) {
+            print('Page load started: $url');
+          },
+          onLoadStop:
+              (webview.InAppWebViewController controller, Uri? url) async {
+            print('Page load stopped: $url');
+
+            final cookieManager = webview.CookieManager.instance();
+            List<webview.Cookie> cookies =
+                await cookieManager.getCookies(url: url!);
+
+            // Save jwt token
+            await saveToken(
+              cookies.map((c) => Cookie(c.name, c.value)).toList(),
+            );
+
+            // Redirect back to app
+            if (url.toString().startsWith(Routes.ssoRedirect)) {
+              print('Redirect URL detected: $url');
+
+              // Close the web view and go back to the app
+              print('Closing web view and returning to app');
+              Navigator.pop(context);
+            }
+          },
+        ),
+      ),
+    );
+  }
```

Comment:
# This can be ignored 

## This suggestion is just for better readability of the code.

```suggestion
  static Future<void> ssoAuth(BuildContext context) async {
    final loginURL = Uri.parse(Routes.ssoLogin);
    await Navigator.push(context, MaterialPageRoute(builder: (context) => _buildSSOWebView(loginURL, context)));
  }

  static Widget _buildSSOWebView(Uri loginURL, BuildContext context) {
    return webview.InAppWebView(
      initialUrlRequest: webview.URLRequest(url: loginURL),
      onWebViewCreated: (webview.InAppWebViewController controller) {
        debugPrint('Web view created');
      },
      onLoadStart: (webview.InAppWebViewController controller, Uri? url) {
        debugPrint('Page load started: $url');
      },
      onLoadStop: (webview.InAppWebViewController controller, Uri? url) async {
        debugPrint('Page load stopped: $url');
        if (url != null) {
          await _handleSSOLoadStop(url, context);
        }
      },
    );
  }

  static Future<void> _handleSSOLoadStop(Uri url, BuildContext context) async {
    final cookieManager = webview.CookieManager.instance();
    List<webview.Cookie> cookies = await cookieManager.getCookies(url: url);
    bool tokenSaved = await TokenManager.saveToken(cookies.map((c) => Cookie(c.name, c.value)).toList());
    if (tokenSaved && url.toString().startsWith(Routes.ssoRedirect)) {
      debugPrint('Redirect URL detected: $url');
      Navigator.pop(context);
    }
  }
```

Reply:
I'd leave it as it is for now, but will maybe change it as suggested in the future.